lib/ok.ex

```diff
@@ -231,7 +251,21 @@ defmodule OK do
       end
     end
   end
+  defp nest([{:ok, _} = normal | []]) do
+    # last line of the with block is an ok literal
+    quote do
+      unquote(normal)
+    end
+  end
+  
+  defp nest([{{:., _, [{:__aliases__, _, [:OK]}, :success]}, _, _} = normal | []]) do
+    # last line of the with block is an OK.success macro line
+    quote do
+      unquote(normal)
+    end
+  end
```

Comment:
I also wanted to add that if you can come up with a more elegant solution, I think that would be awesome. I'm certainly not as versed in metaprogramming with Elixir as you, and I certainly wouldn't be suprised if you come up with something cool...after all, I think this whole lib is pretty sweet! :candy: :smile: